pkg/process/checks/container_check_test.go

```diff
@@ -0,0 +1,148 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package checks
+
+import (
+	"testing"
+	"time"
+
+	model "github.com/DataDog/agent-payload/v5/process"
+	"github.com/golang/mock/gomock"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	proccontainersmocks "github.com/DataDog/datadog-agent/pkg/process/util/containers/mocks"
+)
+
+// Constant modeled after cacheValidityNoRT to allow for mocking containers functions
+const (
+	cacheValidityNoRTTest = 2 * time.Second
+)
+
+func containerCheckWithMockContainerProvider(t *testing.T) (*ContainerCheck, *proccontainersmocks.MockContainerProvider) {
+	t.Helper()
+	mockCtrl := gomock.NewController(t)
+	containerProvider := proccontainersmocks.NewMockContainerProvider(mockCtrl)
+	sysInfo := &model.SystemInfo{
+		Cpus: []*model.CPUInfo{
+			{CoreId: "1"},
+			{CoreId: "2"},
+			{CoreId: "3"},
+			{CoreId: "4"},
+		},
+	}
+	hostInfo := &HostInfo{
+		SystemInfo: sysInfo,
+	}
+
+	return &ContainerCheck{
+		hostInfo:          hostInfo,
+		containerProvider: containerProvider,
+	}, containerProvider
+}
+
+func TestContainerCheckWithChunking(t *testing.T) {
+	containerCheck, mockContainerProvider := containerCheckWithMockContainerProvider(t)
+
+	// Set small size per chunk to encourage chunking behavior
```

Comment:
[minor][wording] Maybe instead of encourage say 'force'. 'encourage' seems like a suggestion and potentially non-deterministic.